gson/src/main/java/com/google/gson/stream/JsonWriter.java

```diff
@@ -498,6 +498,10 @@ public JsonWriter name(String name) throws IOException {
     if (stackSize == 0) {
       throw new IllegalStateException("JsonWriter is closed.");
     }
+    // As currently in any valid case name function cannot be called if stack top has these 2 statuses
```

Comment:
I found this comment a bit confusing. I think it isn't really necessary anyway? It doesn't really tell the reader anything that they can't already see from the following statement.

Reply:
got it! removed this unecessary comment